File: fontes/backend/php/fenix-doctrine/src/model/nfe/NfeTransporteVolume.php
<?php
/*******************************************************************************
Title: T2Ti ERP Fenix                                                                
Description: Model relacionado à tabela [NFE_TRANSPORTE_VOLUME] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
declare(strict_types=1);

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;

/**
 * @ORM\Entity
 * @ORM\Table(name="NFE_TRANSPORTE_VOLUME")
 */
class NfeTransporteVolume extends ModelBase implements \JsonSerializable
{
	/**
	 * @ORM\Id
	 * @ORM\Column(type="integer")
	 * @ORM\GeneratedValue(strategy="AUTO")
	 */
	private $id;

	/**
	 * @ORM\Column(type="integer", name="QUANTIDADE")
	 */
	private $quantidade;

	/**
	 * @ORM\Column(type="string", name="ESPECIE")
	 */
	private $especie;

	/**
	 * @ORM\Column(type="string", name="MARCA")
	 */
	private $marca;

	/**
	 * @ORM\Column(type="string", name="NUMERACAO")
	 */
	private $numeracao;

	/**
	 * @ORM\Column(type="float", name="PESO_LIQUIDO")
	 */
	private $pesoLiquido;

	/**
	 * @ORM\Column(type="float", name="PESO_BRUTO")
	 */
	private $pesoBruto;


    /**
     * Relations
     */
    
    /**
     * @ORM\OneToMany(targetEntity="NfeTransporteVolumeLacre", mappedBy="nfeTransporteVolume", cascade={"persist", "remove"})
     */
    private $listaNfeTransporteVolumeLacre;

    /**
     * @ORM\ManyToOne(targetEntity="NfeTransporte", inversedBy="listaNfeTransporteVolume")
     * @ORM\JoinColumn(name="ID_NFE_TRANSPORTE", referencedColumnName="id")
     */
    private $nfeTransporte;


    /**
     * Gets e Sets
     */

    public function getId() 
	{
		return $this->id;
	}

	public function setId($id) 
	{
		$this->id = $id;
	}

    public function getQuantidade() 
	{
		return $this->quantidade;
	}

	public function setQuantidade($quantidade) 
	{
		$this->quantidade = $quantidade;
	}

    public function getEspecie() 
	{
		return $this->especie;
	}

	public function setEspecie($especie) 
	{
		$this->especie = $especie;
	}

    public function getMarca() 
	{
		return $this->marca;
	}

	public function setMarca($marca) 
	{
		$this->marca = $marca;
	}

    public function getNumeracao() 
	{
		return $this->numeracao;
	}

	public function setNumeracao($numeracao) 
	{
		$this->numeracao = $numeracao;
	}

    public function getPesoLiquido() 
	{
		return $this->pesoLiquido;
	}

	public function setPesoLiquido($pesoLiquido) 
	{
		$this->pesoLiquido = $pesoLiquido;
	}

    public function getPesoBruto() 
	{
		return $this->pesoBruto;
	}

	public function setPesoBruto($pesoBruto) 
	{
		$this->pesoBruto = $pesoBruto;
	}

    public function getNfeTransporte(): ?NfeTransporte 
	{
		return $this->nfeTransporte;
	}

	public function setNfeTransporte(?NfeTransporte $nfeTransporte) 
	{
		$this->nfeTransporte = $nfeTransporte;
	}

    public function getListaNfeTransporteVolumeLacre() 
	{
		return $this->listaNfeTransporteVolumeLacre->toArray();
	}

	public function setListaNfeTransporteVolumeLacre(array $listaNfeTransporteVolumeLacre) {
		$this->listaNfeTransporteVolumeLacre = new ArrayCollection();
		for ($i = 0; $i < count($listaNfeTransporteVolumeLacre); $i++) {
			$nfeTransporteVolumeLacre = $listaNfeTransporteVolumeLacre[$i];
			$nfeTransporteVolumeLacre->setNfeTransporteVolume($this);
			$this->listaNfeTransporteVolumeLacre->add($nfeTransporteVolumeLacre);
		}
	}


    /**
     * Constructor
     */
    public function __construct($objetoJson)
    {
        if (isset($objetoJson)) {
            isset($objetoJson->id) ? $this->setId($objetoJson->id) : $this->setId(null);
            $this->mapear($objetoJson);
        }

		
		$this->listaNfeTransporteVolumeLacre = new ArrayCollection();
		$listaNfeTransporteVolumeLacreJson = $objetoJson->listaNfeTransporteVolumeLacre;
		if ($listaNfeTransporteVolumeLacreJson != null) {
			for ($i = 0; $i < count($listaNfeTransporteVolumeLacreJson); $i++) {
				$objeto = new NfeTransporteVolumeLacre($listaNfeTransporteVolumeLacreJson[$i]);
				$objeto->setNfeTransporteVolume($this);
				$this->listaNfeTransporteVolumeLacre->add($objeto);
			}
		}

    }

    /**
     * Mapping
     */
    public function mapear($objeto)
    {
        if (isset($objeto)) {
			$this->setQuantidade($objeto->quantidade);
			$this->setEspecie($objeto->especie);
			$this->setMarca($objeto->marca);
			$this->setNumeracao($objeto->numeracao);
			$this->setPesoLiquido($objeto->pesoLiquido);
			$this->setPesoBruto($objeto->pesoBruto);
		}
    }


    /**
     * Validation
     */
    public function validarObjetoRequisicao($objJson, $id)
    {
        return parent::validarObjeto($objJson, $id, 'NfeTransporteVolume');
    }


    /**
     * Serialization
     * {@inheritdoc}
     */
    public function jsonSerialize()
    {
        return [
			'id' => $this->getId(),
			'quantidade' => $this->getQuantidade(),
			'especie' => $this->getEspecie(),
			'marca' => $this->getMarca(),
			'numeracao' => $this->getNumeracao(),
			'pesoLiquido' => $this->getPesoLiquido(),
			'pesoBruto' => $this->getPesoBruto(),
			'listaNfeTransporteVolumeLacre' => $this->getListaNfeTransporteVolumeLacre(),
        ];
    }
}

File: fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/model/nfe/NfeTransporteVolume.java
/*******************************************************************************
Title: T2Ti ERP Fenix                                                                
Description: Model relacionado à tabela [NFE_TRANSPORTE_VOLUME] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
package com.t2ti.fenix.model.nfe;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import java.math.BigDecimal;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import java.util.Set;
import javax.persistence.OneToMany;
import com.fasterxml.jackson.annotation.JsonIgnore;
import javax.persistence.CascadeType;

@Entity
@Table(name="NFE_TRANSPORTE_VOLUME")
@NamedQuery(name="NfeTransporteVolume.findAll", query="SELECT t FROM NfeTransporteVolume t")
public class NfeTransporteVolume implements Serializable {
	private static final long serialVersionUID = 1L;

    @Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

    @Column(name="QUANTIDADE")
	private Integer quantidade;

    @Column(name="ESPECIE")
	private String especie;

    @Column(name="MARCA")
	private String marca;

    @Column(name="NUMERACAO")
	private String numeracao;

    @Column(name="PESO_LIQUIDO")
	private BigDecimal pesoLiquido;

    @Column(name="PESO_BRUTO")
	private BigDecimal pesoBruto;

    @OneToMany(mappedBy = "nfeTransporteVolume", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<NfeTransporteVolumeLacre> listaNfeTransporteVolumeLacre;

    @ManyToOne
	@JsonIgnore
	@JoinColumn(name="ID_NFE_TRANSPORTE")
    private NfeTransporte nfeTransporte;

	public NfeTransporteVolume() {
	}

    public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

    public Integer getQuantidade() {
		return this.quantidade;
	}

	public void setQuantidade(Integer quantidade) {
		this.quantidade = quantidade;
	}

    public String getEspecie() {
		return this.especie;
	}

	public void setEspecie(String especie) {
		this.especie = especie;
	}

    public String getMarca() {
		return this.marca;
	}

	public void setMarca(String marca) {
		this.marca = marca;
	}

    public String getNumeracao() {
		return this.numeracao;
	}

	public void setNumeracao(String numeracao) {
		this.numeracao = numeracao;
	}

    public BigDecimal getPesoLiquido() {
		return this.pesoLiquido;
	}

	public void setPesoLiquido(BigDecimal pesoLiquido) {
		this.pesoLiquido = pesoLiquido;
	}

    public BigDecimal getPesoBruto() {
		return this.pesoBruto;
	}

	public void setPesoBruto(BigDecimal pesoBruto) {
		this.pesoBruto = pesoBruto;
	}

    public NfeTransporte getNfeTransporte() {
		return this.nfeTransporte;
	}

	public void setNfeTransporte(NfeTransporte nfeTransporte) {
		this.nfeTransporte = nfeTransporte;
	}

    public Set<NfeTransporteVolumeLacre> getListaNfeTransporteVolumeLacre() {
		return this.listaNfeTransporteVolumeLacre;
	}

	public void setListaNfeTransporteVolumeLacre(Set<NfeTransporteVolumeLacre> listaNfeTransporteVolumeLacre) {
		this.listaNfeTransporteVolumeLacre = listaNfeTransporteVolumeLacre;
		for (NfeTransporteVolumeLacre nfeTransporteVolumeLacre : listaNfeTransporteVolumeLacre) {
			nfeTransporteVolumeLacre.setNfeTransporteVolume(this);
		}
	}

		
}